geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerFeatureSource.scala

```diff
@@ -80,20 +88,21 @@ class KafkaConsumerFeatureSource(entry: ContentEntry,
   def processNewFeatures(update: CreateOrUpdate): Unit = {
     val sf = update.f
     val id = update.id
-    Option(features.get(id)).foreach {  old => qt.remove(old.env, old.sf) }
+    Option(features.asMap().get(id)).foreach {  old => qt.remove(old.env, old.sf) }
```

Comment:
It would probably be better to use features.getIfPresent() - asMap might have some hidden cost we can avoid. Applies to the other uses of asMap too.